scene/2d/tile_map.cpp

```diff
@@ -2668,19 +2668,25 @@ Rect2 TileMap::get_used_rect() { // Not const because of cache
 	// Return the rect of the currently used area
 	if (used_size_cache_dirty) {
 		bool first = true;
+		used_size_cache = Rect2();
+
 		for (unsigned int i = 0; i < layers.size(); i++) {
 			const Map<Vector2i, TileMapCell> &tile_map = layers[i].tile_map;
 			if (tile_map.size() > 0) {
 				if (first) {
-					used_size_cache = Rect2(tile_map.front()->key().x, tile_map.front()->key().y, 1, 1);
+					used_size_cache = Rect2(tile_map.front()->key().x, tile_map.front()->key().y, 0, 0);
 					first = false;
 				}
 
 				for (Map<Vector2i, TileMapCell>::Element *E = tile_map.front(); E; E = E->next()) {
-					used_size_cache.expand_to(Vector2(E->key().x + 1, E->key().y + 1));
+					used_size_cache.expand_to(Vector2(E->key().x, E->key().y));
 				}
 			}
 		}
+
+		if (!first) { // first is true if every layer is empty.
+			used_size_cache.expand_to(used_size_cache.get_end() + Vector2(1, 1)); // The cache expands to top-left coordinate, so we add one full tile.
```

Comment:
Maybe `used_size_cache.size += Vector2i(1, 1)` would be clearer instead ? (And should be Vector2i btw)